run/tesla_api.py

```diff
@@ -0,0 +1,397 @@
+######################################
+def get_odometer():
+    data = get_vehicle_state()
+    return int(data['response']['odometer'])
+
+
+def is_car_locked():
+    data = get_vehicle_state()
+    return data['response']['locked']
+
+
+######################################
+# API POST Functions
+######################################
+def wake_up_vehicle():
+    # It's not really an error, but I want to make sure this always prints,
+    # even if DEBUG is False. Also I need the timestamp.
+    _error('Sending wakeup API command...')
+    result = _execute_request(
+        '{}/{}/wake_up'.format(base_url, tesla_api_json['vehicle_id']),
+        method='POST'
+    )
+    return result['response']['state']
+
+
+def set_charge_limit(percent):
+    return _execute_request(
+        '{}/{}/command/set_charge_limit'.format(base_url, tesla_api_json['vehicle_id']),
+        method='POST',
+        data={'percent': percent}
+    )
+
+
+######################################
+# MAIN
+######################################
+def main():
+    # Build the list of available Tesla API function names by getting the
+    # callables from globals() and skipping the non-API functions.
+    non_api_names = ['main', 'pprint']
+    function_names = []
+    for name, func in globals().items():
+        if (callable(func)
+            and not name.startswith('_')
+            and not name in non_api_names
+        ):
+            function_names.append(name)
+    function_names_string = '\n'.join(function_names)
+
+    # Parse the CLI arguments.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'function',
+        help="The name of the function to run. Available functions are:\n {}".format(function_names_string))
+    parser.add_argument(
+        '--arguments',
+        help="Add arguments to the function by passing comma-separated key:value pairs."
+    )
+    parser.add_argument(
+        "--debug",
+        action="store_true",
+        help="Print debug output."
+    )
+    args = parser.parse_args()
+
+    SETTINGS['DEBUG'] = args.debug
+
+    # We call this now so DEBUG will be set correctly.
+    _load_tesla_api_json()
+
+    # Apply any arguments that the user may have provided.
+    kwargs = {}
+    if args.arguments:
+        for kwarg_string in [arg.strip() for arg in args.arguments.split(',')]:
+            key, value = kwarg_string.split(':')
+            kwargs[key] = value
+    # Render the arguments as a POST body.
+    kwargs_string = ''
+    if kwargs:
+        kwargs_string = ', '.join(
+            '{}={}'.format(key, value) for key, value in kwargs.items()
+        )
+
+    # Retrieve the setup variables, which is where the account credentials
+    # or bearer token, and the VIN are stored. Since teslausb_setup_variables.conf
+    # is actually a shell script, rather than a ConfigParser file, we have to
+    # parse it manually.
+    with open('/root/teslausb_setup_variables.conf', 'r') as conf_file:
```

Comment:
it looks like it does quite a bit of work above before it even gets to parsing the .conf file, and then after it parses the .conf, it doesn't appear to check whether any of the variables were actually set.
What will happen when the user didn't configure it to use the Tesla API to keep the car awake?

Reply:
It checks if the variables were set in `_get_api_token()`. And it kills the script if the variables weren't set. See line 127 and 128.

Reply:
But doesn't that only happen until it actually calls that function, from line 391 below? It's doing quite a bit of work before that (not the least of which is starting up the python interpreter). I realize it will only do that every 5 minutes, but I'd still like to avoid it if it is not needed.